server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/JobEventHandler.java

```diff
@@ -224,4 +223,18 @@ private Long getPodStartTime(String jobId) {
         }
         return startTimes.stream().min(Long::compareTo).get();
     }
+
+    private String getPodFailedReason(String jobId) {
+        List<V1Pod> pods = k8sClient.getPodsByJobNameQuietly(jobId);
+
+        var failedReasons = pods.stream().filter(p -> p.getStatus() != null)
+                .filter(p -> Objects.equals(p.getStatus().getPhase(), "Failed"))
+                .map(p -> joinReasons(p.getStatus().getReason(), p.getStatus().getMessage()))
+                .filter(Objects::nonNull).collect(Collectors.toList());
+
+        if (failedReasons.size() == 0) {
+            return null;
+        }
+        return failedReasons.stream().min(String::compareTo).get();
```

Comment:
Is it fit to get reason by the first char of failed reason?

Reply:
On the native kubelet, the error of the first pod is usually very representative, such as the image does not exist, etc.

If the pod is up, the user can check the cause of the failure directly from the Pod log, which is usually a problem with the user's own program.

Reply:
If there are more than one pod fail, just concat them?

Reply:
All added